packages/orchestrator/lib/routes/v1/search.ts

```diff
@@ -0,0 +1,56 @@
+import { z } from 'zod';
+import type { Scheduler, Task } from '@nangohq/scheduler';
+import type { ApiError, Endpoint } from '@nangohq/types';
+import type { EndpointRequest, EndpointResponse, RouteHandler, Route } from '@nangohq/utils';
+import { validateRequest } from '@nangohq/utils';
+
```

Comment:
What is the purpose of search in the context of the processor?

Reply:
It is the search task endpoint. The processor must be able to check if pending tasks have been cancelled so it need to be able to search for specific tasks. 
This endpoint could also be used in internal tooling to display list of tasks per queue/state/etc...